packages/api/srcmd/encoding.mts

```diff
@@ -0,0 +1,90 @@
+import { marked } from 'marked';
+import type {
+  CodeCellType,
+  MarkdownCellType,
+  PackageJsonCellType,
+  TitleCellType,
+  SrcbookMetadataType,
+  CellWithPlaceholderType,
+  PlaceholderCellType,
+} from '@srcbook/shared';
+
+marked.use({ gfm: true });
+
+export function encode(
+  allCells: CellWithPlaceholderType[],
+  metadata: SrcbookMetadataType,
+  options: { inline: boolean },
+) {
+  const [firstCell, secondCell, ...remainingCells] = allCells;
+  const titleCell = firstCell as TitleCellType;
+  const packageJsonCell = secondCell as PackageJsonCellType;
+  const cells = remainingCells as (MarkdownCellType | CodeCellType | PlaceholderCellType)[];
+
+  const encoded = [
+    `<!-- srcbook:${JSON.stringify(metadata)} -->`,
+    encodeTitleCell(titleCell),
+    encodePackageJsonCell(packageJsonCell, options),
+    ...cells.map((cell) => {
+      switch (cell.type) {
+        case 'code':
+          return encodeCodeCell(cell, options);
+        case 'markdown':
+          return encodeMarkdownCell(cell);
+        case 'placeholder':
+          return encodePlacebolderCell(cell);
```

Comment:
@nichochar there shouldn't be a coupling here to AI-specific things, so we can instead represent a placeholder cell as a cell and then render whatever the contents of the cell's text is.

Reply:
This also allows it to be used for other things in the future if needed.